### PR TITLE
bpo-26589: Add http status code 451

### PR DESCRIPTION
**Pull request by rhettinger**
Previously given LGTM by R David Murray on 5/29/2016

https://bugs.python.org/issue26589

### CODE REVIEW

File: Lib/http/__init__.py [hunk 1]
```diff
@@ -114,6 +115,10 @@ def __new__(cls, value, phrase, description=''):
         'Request Header Fields Too Large',
         'The server is unwilling to process the request because its header '
         'fields are too large')
+    UNAVAILABLE_FOR_LEGAL_REASONS = (451,
```
Comment (tirkarthi) [inline Lib/http/__init__.py:118]: Please add this constant to list of tested constants like the previously added constant MISDIRECTED_REQUEST : https://github.com/python/cpython/pull/2589/files#diff-4334c6b7e8fcfc88f2555f65fa7ab337R1358

Comment (rhettinger) [reply Lib/http/__init__.py:118]: Done.

File: Doc/library/http.rst [hunk 2]
```diff
@@ -106,6 +106,7 @@ Code    Enum Name                           Details
 ``428`` ``PRECONDITION_REQUIRED``           Additional HTTP Status Codes :rfc:`6585`
 ``429`` ``TOO_MANY_REQUESTS``               Additional HTTP Status Codes :rfc:`6585`
 ``431`` ``REQUEST_HEADER_FIELDS_TOO_LARGE`` Additional HTTP Status Codes :rfc:`6585`
+``451`` ``UNAVAILABLE_FOR_LEGAL_REASONS``   Status Code to Report Legal Obstacles :rfc:`7725`
```
Comment (epicfaace) [inline Doc/library/http.rst]: Perhaps you should change it to the title of the rfc for consistency?
```suggestion
``451`` ``UNAVAILABLE_FOR_LEGAL_REASONS``   An HTTP Status Code to Report Legal Obstacles :rfc:`7725`
```

File: Doc/library/http.rst [hunk 3]
```diff
@@ -126,3 +127,6 @@ equal to the constant name (i.e. ``http.HTTPStatus.OK`` is also available as
 
 .. versionchanged:: 3.7
    Added ``421 MISDIRECTED_REQUEST`` status code.
+
+.. versionadded:: 3.8
```
Comment (epicfaace) [inline Doc/library/http.rst:131]: Should this be:
```suggestion
.. versionadded:: 3.9
```

File: Lib/http/__init__.py [hunk 4]
```diff
@@ -15,6 +15,7 @@ class HTTPStatus(IntEnum):
         * RFC 7238: Permanent Redirect
         * RFC 2295: Transparent Content Negotiation in HTTP
         * RFC 2774: An HTTP Extension Framework
+        * RFC 7725: Status Code to Report Legal Obstacles
```
Comment (epicfaace) [inline Lib/http/__init__.py]:
```suggestion
        * RFC 7725: An HTTP Status Code to Report Legal Obstacles
```

File: Doc/library/http.rst (hunk 3 above)
Comment (rhettinger) [reply Doc/library/http.rst:131]: I'm backporting this to 3.8

File: Doc/library/http.rst (hunk 2 above)
Comment (rhettinger) [reply Doc/library/http.rst]: Okay.

File: Lib/http/__init__.py (hunk 4 above)
Comment (rhettinger) [reply Lib/http/__init__.py]: Okay

**Review by tirkarthi — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**